services/121-service/src/payments/fsp-integration/safaricom/safaricom.service.ts

```
@@ -91,6 +90,11 @@ export class SafaricomService
       (user) => user.referenceId == jobData.paPaymentData.referenceId,
     );
 
+    if (!resultUser) {
+      // XXX: should throw error?
```

I think, Yes, its better to throw an error here rather than just return.